src/datajudge/constraints/miscs.py

```diff
@@ -119,6 +119,25 @@ def test(self, engine: sa.engine.Engine) -> TestResult:
         return TestResult.failure(assertion_text)
 
 
+class FunctionalDependency(Constraint):
+    def __init__(self, ref: DataReference, key_columns: List[str], **kwargs):
+        super().__init__(ref, ref_value=object(), **kwargs)
+        self.key_columns = key_columns
+
+    def test(self, engine: sa.engine.Engine) -> TestResult:
+        violations, _ = db_access.get_functional_dependency_violations(
+            engine, self.ref, self.key_columns
+        )
+        if not violations:
+            return TestResult.success()
+
+        assertion_text = (
```

Comment:
The assertion text is great! Would it make sense to limit the amount of violations used for the text? If something has gone terribly wrong in a database, this could cram the console.

Reply:
I limited the output to 5 violations